java-impl/src/main/java/org/jetbrains/research/refactorinsight/data/methods/SplitConditionalJavaHandler.java

```diff
@@ -1,13 +1,48 @@
 package org.jetbrains.research.refactorinsight.data.methods;
 
+import gr.uom.java.xmi.decomposition.AbstractCodeFragment;
+import gr.uom.java.xmi.diff.SplitConditionalRefactoring;
+import org.jetbrains.research.refactorinsight.data.Group;
 import org.jetbrains.research.refactorinsight.data.JavaRefactoringHandler;
 import org.jetbrains.research.refactorinsight.data.RefactoringInfo;
 import org.refactoringminer.api.Refactoring;
 
+import static org.jetbrains.research.refactorinsight.data.util.JavaUtils.calculateSignatureForVariableDeclarationContainer;
+import static org.jetbrains.research.refactorinsight.data.util.JavaUtils.createCodeRangeFromJava;
+
 public class SplitConditionalJavaHandler extends JavaRefactoringHandler {
     @Override
     public RefactoringInfo specify(Refactoring refactoring, RefactoringInfo info) {
-        return null;
+        SplitConditionalRefactoring ref = (SplitConditionalRefactoring) refactoring;
+
+        ref.getSplitConditionals().forEach(conditional ->
+                info.addMarking(createCodeRangeFromJava(ref.getOriginalConditional().codeRange()),
+                        createCodeRangeFromJava(conditional.codeRange()), true));
+
+        String conditionalString = ref.getOriginalConditional().getString();
+        String oldConditional = (conditionalString.contains("\n") ? conditionalString.substring(0,
+                conditionalString.indexOf("\n")) : conditionalString);
+
+        StringBuilder splitConditionals = new StringBuilder();
+        int i = 0;
+        for(AbstractCodeFragment splitConditional : ref.getSplitConditionals()) {
```

Comment:
Maybe we can extract this code into a separate method? It's used in `SplitConditionalJavaHandler`, `MergeConditionalJavaHandler`, and `MergeCatchJavaHandler`